x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/preview_inventory_metric_threshold_alert.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { first } from 'lodash';
+import { InventoryMetricConditions } from './types';
+import { IScopedClusterClient } from '../../../../../../../src/core/server';
+import { InfraSource } from '../../../../common/http_api/source_api';
+import { getIntervalInSeconds } from '../../../utils/get_interval_in_seconds';
+import { InventoryItemType } from '../../../../common/inventory_models/types';
+import { evaluateCondition } from './evaluate_condition';
+
+interface InventoryMetricThresholdParams {
+  criteria: InventoryMetricConditions[];
+  filterQuery: string | undefined;
+  nodeType: InventoryItemType;
+  sourceId?: string;
+}
+
+interface PreviewInventoryMetricThresholdAlertParams {
+  callCluster: IScopedClusterClient['callAsCurrentUser'];
+  params: InventoryMetricThresholdParams;
+  config: InfraSource['configuration'];
+  lookback: 'h' | 'd' | 'w' | 'M';
```

Comment:
nit: Looks like this is used often, maybe it should be an exported type somewhere? Or, maybe we can use https://github.com/elastic/kibana/blob/a95ae8c037b77caeb85158b6de513cb5b5ef3e55/packages/elastic-datemath/src/index.d.ts#L21